docs: general update of documentation (#460)
jina-ai/finetuner: docs/2_step_by_step/2_5_choose_backbone.md

```

+ ResNets are suitable for image-to-image search tasks with high performance requirement.
+ EfficientNets are suitable for image-to-image search tasks with fast training and inference. The model is more light-weighted than ResNet.

> + CLIP is the one for text-to-image search and the image  do not necessary to have any text descriptors.

Reviewer: ```suggestion
+ CLIP is the one for text-to-image search, where the images do not need to have any text descriptors.
```
